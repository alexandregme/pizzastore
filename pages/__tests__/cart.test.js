import React from 'react';
import {shallow} from 'enzyme';
import expect from 'expect';
import Cart from '../cart';
import Layout from '../../components/layout';

describe("Cart page", ()=>{
    let mountedCart;

    beforeEach(()=>{
        mountedCart = shallow(<Cart />);
    });

    it('renders without crashing', () => {
        shallow(<Cart />);
        expect(mountedCart).toMatchSnapshot();
    });

    it('expect to have Layout component', () => {
        expect(mountedCart.find(Layout)).toHaveLength(1);
    });
});