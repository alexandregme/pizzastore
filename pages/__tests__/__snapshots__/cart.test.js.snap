// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart page renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cart />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <p>
        Cart Page
      </p>,
      "title": "Cart Page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Cart Page",
      },
      "ref": null,
      "rendered": "Cart Page",
      "type": "p",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <p>
          Cart Page
        </p>,
        "title": "Cart Page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cart Page",
        },
        "ref": null,
        "rendered": "Cart Page",
        "type": "p",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
